Content.Shared/Chemistry/Components/MovespeedModifierMetabolismComponent.cs

```diff
@@ -0,0 +1,86 @@
+using Content.Shared.Movement.Components;
+using Robust.Shared.GameObjects;
+using Robust.Shared.Players;
+using Robust.Shared.Serialization;
+using Robust.Shared.ViewVariables;
+using System;
+using Robust.Shared.GameStates;
+using Robust.Shared.Timing;
+using Robust.Shared.IoC;
+
+namespace Content.Shared.Chemistry.Components
+{
+    //TODO: refactor movement modifier component because this is a pretty poor solution
+    [RegisterComponent]
+    [NetworkedComponent]
+    public sealed class MovespeedModifierMetabolismComponent : Component, IMoveSpeedModifier
+    {
+        [Dependency] private readonly IGameTiming _gameTiming = default!;
+
+        [ViewVariables]
+        public override string Name => "MovespeedModifierMetabolismComponent";
+
+        [ViewVariables]
+        public float WalkSpeedModifier { get; set; }
+
+        [ViewVariables]
+        public float SprintSpeedModifier { get; set; }
+
+        [ViewVariables]
+        public int EffectTime { get; set; }
+
+        public (TimeSpan Start, TimeSpan End)? ModifierTimer { get; set; }
+
+        public void ResetModifiers()
+        {
+            WalkSpeedModifier = 1;
+            SprintSpeedModifier = 1;
+
+            if (Owner.TryGetComponent(out MovementSpeedModifierComponent? modifier))
+            {
+                modifier.RefreshMovementSpeedModifiers();
+            }
+            Dirty();
+        }
+
+        public void Update(float delta)
+        {
+            var curTime = _gameTiming.CurTime;
+
+            if (ModifierTimer != null)
+            {
+                if (ModifierTimer.Value.End <= curTime)
+                {
+                    ModifierTimer = null;
+                    ResetModifiers();
+                    Dirty();
+                }
+            }
+        }
+
+        /// <summary>
+        /// Perpetuate the modifiers further.
+        /// </summary>
+
+        public override ComponentState GetComponentState(ICommonSession player)
+        {
+            return new MovespeedModifierMetabolismComponentState(WalkSpeedModifier, SprintSpeedModifier, ModifierTimer);
+        }
+
+        [Serializable, NetSerializable]
+        public class MovespeedModifierMetabolismComponentState : ComponentState
+        {
+            public float WalkSpeedModifier { get; }
+            public float SprintSpeedModifier { get; }
+            public (TimeSpan Start, TimeSpan End)? ModifierTimer { get; }
+
+            public MovespeedModifierMetabolismComponentState(float walkSpeedModifier, float sprintSpeedModifier, (TimeSpan Start, TimeSpan End)? modifierTimer)
+            {
+                WalkSpeedModifier = walkSpeedModifier;
+                SprintSpeedModifier = sprintSpeedModifier;
+                ModifierTimer = modifierTimer;
+            }
+        }
+    }
```

Comment:
1. ECS plz (at least everything but the state).
2. I don't think you need the start timer but I'd need to verify in-game with it.
3. Please use events so you only update the components that are active
4. Some of this is duplicate with the system

Reply:
1. okay I think I've ECS'd everything
2. OK I've removed that, since it was unnecessary and I never checked it
3. kinda unsure what you mean by this
4. yeah I got rid of that